eduNEXT/eox-nelp
# feat: add xapi actor filter

## Description

This allows to modify the actor field of event-routing-backends https://github.com/nelc/event-routing-backends/blob/and/add_openedx_filters/event_routing_backends/processors/xapi/transformer.py#L82

## Testing instructions
1. Use this version of event routing https://github.com/nelc/event-routing-backends/pull/5
2. add the settings        
```
       XAPI_AGENT_IFI_TYPE = "mbox"
       OPEN_EDX_FILTERS_CONFIG = {
            "event_routing_backends.processors.xapi.transformer.xapi_transformer.get_actor": {
                "pipeline": ["eox_nelp.openedx_filters.xapi.filters.XApiActorFilter"],
                "fail_silently": False,
            }
        }
```
3. Raise a valid event like, show_answer in a unit 
4.  Check the result, in my case I'm using aspects, so I can check that in the superset panel


### Before
![image](https://github.com/eduNEXT/eox-nelp/assets/36200299/1a6772aa-0251-4840-983e-93be2d84514e)

### After
![image](https://github.com/eduNEXT/eox-nelp/assets/36200299/51fc1aa2-4380-43d1-8a05-4a47183db8d7)

## Additional information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, security, or accessibility.
- Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.

## Checklist for Merge
